src/AggregateFunctions/AggregateFunctionUniqHyperLogLogPlusPlus.h

```diff
@@ -0,0 +1,3072 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/IDataType.h>
+#include <DataTypes/DataTypeAggregateFunction.h>
+#include <Common/FieldVisitorConvertToNumber.h>
+#include <Common/FieldVisitors.h>
+#include <Common/HashTable/Hash.h>
+#include <Common/HyperLogLogWithSmallSetOptimization.h>
+#include <Parsers/NullsAction.h>
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int PARAMETER_OUT_OF_BOUND;
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+extern const int INCORRECT_DATA;
+}
+
+struct HyperLogLogPlusPlusData
```

Comment:
The existing HyperLogLog-based `uniqHLL12` is implemented in src/AggregateFunction/AggregateFunctionUniq.h:

```
/// uniqHLL12

template <typename T, bool is_able_to_parallelize_merge_>     ■ Invalid case style for
struct AggregateFunctionUniqHLL12Data
{
    using Set = HyperLogLogWithSmallSetOptimization<T, 16, 12>;
    Set set;

    constexpr static bool is_able_to_parallelize_merge = is_able_to_parallelize_merge_;
    constexpr static bool is_parallelize_merge_prepare_needed = false;
    constexpr static bool is_variadic = false;

    static String getName() { return "uniqHLL12"; }
};
```

Can we not copy-paste the logic but instead integrate the HLL++ somehow into the existing code?

Reply:
@rschu1ze I'm on a vacation recently. I'll think about it after that.

Reply:
@rschu1ze I'm back..  Your suggestion sounds good. I'll try it in recent 2 weeks.